src/classes/PS_CommitmentRequest.cls

```diff
@@ -416,6 +437,38 @@ public inherited sharing class PS_CommitmentRequest {
             return this;
         }
 
+        public RequestBody withPaymentMethodType(String paymentMethodType) {
+            switch on paymentMethodType {
+                when 'ACH' {
```

Comment:
These strings represent the RD2 payment method field's picklist values.

Reply:
I think we can do constants, but then we'd need to lose the switch statement because switch conditions need to be  known at run time, and static final vars don't qualify sadly. I'll move these strings into RD_Constants, and I think they'll be called `RD2_PAYMENT_METHOD_ACH` and `RD2_PAYMENT_METHOD_CREDIT_CARD`, and then this method will be an `if` statement rather than a `switch`.

Reply:
Interesting, I didn't know that about `switch`, I'm fine either way!